src/darwin/Framework/CHIP/MTRDeviceControllerFactory.mm

```diff
@@ -822,50 +830,26 @@ - (MTRDeviceController * _Nullable)createControllerOnNewFabric:(MTRDeviceControl
                                   error:error];
 }
 
-- (MTRDeviceController * _Nullable)createController:(MTRDeviceControllerStartupParameters *)startupParameters
-                                              error:(NSError * __autoreleasing *)error
-{
-    [self _assertCurrentQueueIsNotMatterQueue];
-
-    return [self _startDeviceController:startupParameters
-                          fabricChecker:^MTRDeviceControllerStartupParamsInternal *(
-                              FabricTable * fabricTable, MTRDeviceController * controller, CHIP_ERROR & fabricError) {
-                              auto advertiseOperational = self.advertiseOperational && startupParameters.shouldAdvertiseOperational;
-                              auto * params =
-                                  [[MTRDeviceControllerStartupParamsInternal alloc] initForNewController:controller
-                                                                                             fabricTable:fabricTable
-                                                                                                keystore:self->_keystore
-                                                                                    advertiseOperational:advertiseOperational
-                                                                                                  params:startupParameters
-                                                                                                   error:fabricError];
-                              if (params != nil) {
-                                  if (params.productAttestationAuthorityCertificates == nil) {
-                                      params.productAttestationAuthorityCertificates = self.productAttestationAuthorityCertificates;
-                                  }
-                                  if (params.certificationDeclarationCertificates == nil) {
-                                      params.certificationDeclarationCertificates = self.certificationDeclarationCertificates;
-                                  }
-                              }
-                              return params;
-                          }
-                                  error:error];
-}
-
-- (MTRDeviceController * _Nullable)_createController:(id<MTRDeviceControllerStorageDelegate> _Nullable)storageDelegate
+- (MTRDeviceController * _Nullable)_createController:(nullable MTRDeviceController *)existingController
```

Comment:
Maybe this could also become a `_startDeviceController:...` overload now, and be made so a `MTRDeviceController` is always passed in (and just return BOOL), so the alloc can move to the top-level `createController...` entry points?

Reply:
That seems plausible.  Let me do that.